src/index.ts

```diff
@@ -76,30 +76,35 @@ export default class Onboarding {
     return undefined;
   }
 
-  async _onMessageFromForwarder (event: MessageEvent) {
-    if (this.state === ONBOARDING_STATE.RELOADING) {
-      console.debug('Ignoring message while reloading');
-    } else if (this.state === ONBOARDING_STATE.NOT_INSTALLED) {
-      console.debug('Reloading now to register with MetaMask');
-      this.state = ONBOARDING_STATE.RELOADING;
-      return location.reload();
-    } else if (this.state === ONBOARDING_STATE.INSTALLED) {
-      console.debug('Registering with MetaMask');
-      this.state = ONBOARDING_STATE.REGISTERING;
-      await Onboarding._register();
-      this.state = ONBOARDING_STATE.REGISTERED;
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      event.source?.postMessage({ type: 'metamask:registrationCompleted' }, event.origin);
-      this.stopOnboarding();
-    } else if (this.state === ONBOARDING_STATE.REGISTERING) {
-      console.debug('Already registering - ignoring reload message');
-    } else if (this.state === ONBOARDING_STATE.REGISTERED) {
-      console.debug('Already registered - ignoring reload message');
-    } else {
-      throw new Error(`Unknown state: '${this.state}'`);
+  //TODO: Ask about what type is the 'event' parameter
+  async _onMessageFromForwarder (event: any) {
```

Comment:
I believe this is a `MessageEvent`, similar to the `_onMessage` method above.

```suggestion
  async _onMessageFromForwarder (event: MessageEvent) {
```

I think we can drop the `ts-ignore`, as you've done, by adding a guard around the `postMessage` call, e.g.:

```ts
if (event.source instanceof Window) {
  event.source.postMessage({ type: 'metamask:registrationCompleted' }, event.origin);
}
```